File: problems-extend-adapt/2901-3000/2914_fewest-flips-for-uniform-pairs/solution.java
class Solution {

    public int fewestFlips(String s) {
        // Every beautiful partition refines into length-2 uniform blocks:
        // split each even uniform part down to pairs. So the answer is the
        // number of aligned pairs that are not already uniform, and each
        // such pair costs exactly one change (align both to one value).
        int changes = 0;
        for (int i = 0; i + 1 < s.length(); i += 2) {
            if (s.charAt(i) != s.charAt(i + 1)) ++changes;
        }
        return changes;
    }
}

File: problems-extend-adapt/2901-3000/2914_fewest-flips-for-uniform-pairs/solution.rs
impl Solution {
    pub fn fewest_flips(s: String) -> i32 {
        // Every beautiful partition refines into length-2 uniform blocks:
        // split each even uniform part down to pairs. So the answer is the
        // number of aligned pairs that are not already uniform, and each
        // such pair costs exactly one change (align both to one value).
        let bytes = s.as_bytes();
        let mut changes = 0i32;
        let mut i = 0;
        while i + 1 < bytes.len() {
            if bytes[i] != bytes[i + 1] {
                changes += 1;
            }
            i += 2;
        }
        changes
    }
}
